src/QsCompiler/Transformations/CallGraph.cs

```diff
@@ -54,6 +55,72 @@ internal static bool TryCombineTypeResolutionsForTarget(QsQualifiedName target,
             return success;
         }
 
+        /// <summary>
+        /// Reversed the dependencies of type parameters resolving to other type parameters in the given
+        /// dictionary to create a lookup whose keys are type parameters and whose values are all the type
+        /// parameters that can be updating by knowing the resolution of the lookup's associated key.
+        /// </summary>
+        private static ILookup<Tuple<QsQualifiedName, NonNullable<string>>, Tuple<QsQualifiedName, NonNullable<string>>> GetReplaceable(TypeParameterResolutions.Builder typeParamResolutions)
+        {
+             return typeParamResolutions
+                .Select(kvp => (kvp.Key, GetTypeParameters.Apply(kvp.Value)))   // Get any type parameters in the resolution type.
+                .SelectMany(tup => tup.Item2.Select(value => (tup.Key, value))) // For each type parameter found, match it to the dictionary key.
+                .ToLookup(                                                      // Reverse the keys and resulting type parameters to make the lookup.
+                    kvp => kvp.value,
+                    kvp => kvp.Key);
+        }
+
+        /// <summary>
+        /// Applies all the type parameter resolutions in a dictionary to the whole dictionary,
+        /// resulting in a resolution dictionary that has type parameter keys that are not
+        /// referenced in the values of the dictionary.
+        /// </summary>
+        internal static bool TryCombineTypeResolutionDictionary(out TypeParameterResolutions combined, TypeParameterResolutions resolutionDictionary)
```

Comment:
I think it would make it a bit clearer if the parameters' names were `combinedResolutions` and `detachedResolutions`/`independentResolutions`.

Reply:
This is a good suggestion and I've applied it here as well as to the similar `TryCombineTypeResolutions` method.

Reply:
Thanks!